test_scripts/Polices/External_UCS/External_Consent_ON/012_ATF_Policies_External_Consent_ON_through_ignition_cycles_user_disallowed_RPCs.lua

```diff
@@ -0,0 +1,170 @@
+  -- Main check:
+  -- RPC of Group001 is disallowed to process.
+  --------------------------------------------------------------------------
+  Test[TEST_NAME_ON .. "MainCheck_RPC_of_Group001_is_disallowed"] = function(self)
+    local corid = self.mobileSession:SendRPC("SubscribeVehicleData",{rpm = true})
+    self.mobileSession:ExpectResponse(corid, {success = false, resultCode = "USER_DISALLOWED"})
+    EXPECT_NOTIFICATION("OnHashChange")
+    :Times(0)
+  end
+end
+
+local function IgnitionOffOnActivateApp(test_case_name)
+  common_steps:IgnitionOff("Precondition_Ignition_Off_" .. test_case_name)
+  common_steps:IgnitionOn("Precondition_Ignition_On_" .. test_case_name)
+  common_steps:AddMobileSession("Precondition_Add_Mobile_Session_1_" .. test_case_name, "mobileConnection", "mobileSession")
+  common_steps:RegisterApplication("Precondition_Register_Application_1_" .. test_case_name, "mobileSession", config.application1.registerAppInterfaceParams)
+  common_steps:ActivateApplication("Precondition_Activate_Application_1_" .. test_case_name, config.application1.registerAppInterfaceParams.appName)
+  Test["Precondition_Wait_For_Database_After_Activate_Application_1_" .. test_case_name] = function(self)
+    common_functions:DelayedExp(5000)
+  end
+end
+
+------------------------------------------------------------------------------------------------------
+------------------------------------------ Tests -----------------------------------------------------
+------------------------------------------------------------------------------------------------------
+--------------------------------------------------------------------------
+-- TEST 10:
+-- In case
+-- SDL has received SDL.OnAppPermissionConsent ("externalConsentStatus: ON") from HMI
+-- SDL must
+-- use this value through ignition cycles
+-- until this externalConsentStatus value is changed by corresponding notification from HMI.
+--------------------------------------------------------------------------
+-- Test 10.02:
+-- Description:
+-- 1. disallowed_by_external_consent_entities_on.
+-- 2. HMI->SDL: OnAppPermissionConsent(externalConsentStatus ON).
+-- 3. Ignition Off then On.
+-- 4. Expected Result: externalConsentStatus is kept. RPC is NOT allowed
+
+--------------------------------------------------------------------------
+-- Precondition:
+-- Prepare JSON file with consent groups. Add all consent group names into app_polices of applications
+-- Request Policy Table Update.
+--------------------------------------------------------------------------
+Test[TEST_NAME_ON.."Precondition_Update_Policy_Table"] = function(self)
+  -- create json for PTU from sdl_preloaded_pt.json
+  local data = common_functions_external_consent:ConvertPreloadedToJson()
+  -- insert Group001 into "functional_groupings"
+  data.policy_table.functional_groupings.Group001 = {
+    user_consent_prompt = "ConsentGroup001",
+      disallowed_by_external_consent_entities_on = {{
```

Comment:
@dboltovskyi, the description of script in test set is for disallowed_by_external_consent_entities_off. Please update.